iroh-sync/src/sync.rs

```diff
@@ -786,6 +786,26 @@ impl Record {
     }
 }
 
+/// Progress updates for the streaming set entry by hash operation.
+#[derive(Debug, Serialize, Deserialize)]
+pub enum DocSetProgress {
+    /// Finished adding entry of the given `key` to the document
+    Done {
+        /// Id for this entry, unique for this import
+        id: u64,
+        /// The key of this entry
+        key: Vec<u8>,
```

Comment:
Maybe not for this PR, but we might want to do a newtype for this. This is the key path into the document, correct?